src/thor/trace_route_action.cc

```diff
@@ -189,6 +175,10 @@ thor_worker_t::map_match(valhalla_request_t& request,
     offline_results = matcher->OfflineMatch(trace, best_paths);
   }
 
+  // This exception will be caught upstream and return a 442 error
+  if (offline_results.empty())
```

Comment:
this is also redundant, if you inspect `OfflineMatch` you'll see that there is no code path that can return an empty result set.